### ISSUE TITLE
ERROR IN PLUGIN - check logs for more info - v0.41.0

### ISSUE THREAD

**Issue by pdanpdan** [zellij-org/zellij#3719]
# 2. Issues with the Zellij UI / behavior / crash

## Issue description
After update from prev version I get `ERROR IN PLUGIN - check logs for more info` when zellij starts, with default config.

```
ERROR IN PLUGIN - check logs for more info
┌ status-bar ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│Loading status-bar...                                                                                                                                                      │
│Attempting to load from memory... NOT FOUND                                                                                                                                │
│Attempting to load from cache... NOT FOUND                                                                                                                                 │
│Compiling WASM                                                                                                                                                             │
│ERROR:  failed to start plugin 1 for client 1                                                                                                                              │
│                                                                                                                                                                           │
│Caused by:                                                                                                                                                                 │
│    0: failed to recover cache dir                                                                                                                                         │
│    1: WebAssembly translation error                                                                                                                                       │
│    2: Invalid input WebAssembly code at offset 712955: reference-types not enabled: zero byte expected                                                                    │
│ERROR IN PLUGIN - check logs for more info
```

## Minimal reproduction
- remove all configs for zellij
- start zellij (without any parameter)

## Other relevant information
- using https://github.com/zellij-org/zellij/releases/download/v0.41.0/zellij-x86_64-unknown-linux-musl.tar.gz
- zellij is in `/usr/local/bin`
- PopOS running on Intel CPU
- terminal is wezterm, but the same happens in other terminals
```
Distributor ID: Pop
Description:    Pop!_OS 22.04 LTS
Release:        22.04
Codename:       jammy
```

How can I find the logs with the error?

**Comment by jscarrott** [comment]
Exact same issue on Arch updated via binstall.

**Comment by giyany** [comment]
Hi @pdanpdan, thanks for the swift report. Naturally, this is important.
For general information, zellij logs should write to /$temp_dir/zellij-<UID>/zellij-log/zellij.log. in most systems $temp_dir, is /tmp with some expections (/var/folders/dr/ in MAC).

**Comment by imsnif** [comment]
Looking into this - meanwhile, does this happen with a cargo install on the same machines? Does clearing the cache folder (you can find its location with `zellij setup --check`) help? (beware, the latter will remove all resurrectable sessions from the relevant version)

@bjorn3 - any hints? So far seems to happen on a generic musl install (the plugins are compiled on the CI and inlined in the binary)

**Comment by benjaminbauer** [comment]
same/similar on macOs, install via pre-built binaries:
```
INFO   |zellij_client            | 2024-11-04 10:51:22.483 [main      ] [zellij-client/src/lib.rs:182]: Starting Zellij client! 
INFO   |zellij_server            | 2024-11-04 10:51:22.494 [main      ] [zellij-server/src/lib.rs:522]: Starting Zellij server! 
WARN   |zellij_utils::kdl        | 2024-11-04 10:51:22.497 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file 
WARN   |zellij_utils::kdl        | 2024-11-04 10:51:22.497 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file 
INFO   |zellij_server            | 2024-11-04 10:51:22.498 [main      ] [zellij-server/src/lib.rs:1493]: Compiling plugins using Cranelift 
INFO   |zellij_server::plugins   | 2024-11-04 10:51:22.499 [wasm      ] [zellij-server/src/plugins/mod.rs:232]: Wasm main thread starts 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.500 [server_router] [zellij-server/src/route.rs:1110]: Server not ready, trying to place instruction in retry queue... 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.500 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction. 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.507 [server_router] [zellij-server/src/route.rs:1110]: Server not ready, trying to place instruction in retry queue... 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.507 [server_router] [zellij-server/src/route.rs:1119]: Server not ready, trying to place instruction in retry queue... 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.507 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction. 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.514 [server_router] [zellij-server/src/route.rs:1110]: Server not ready, trying to place instruction in retry queue... 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.514 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction. 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.521 [server_router] [zellij-server/src/route.rs:1119]: Server not ready, trying to place instruction in retry queue... 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.521 [server_router] [zellij-server/src/route.rs:1130]: Server not ready, trying to place instruction in retry queue... 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.522 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction. 
WARN   |zellij_utils::input::layo| 2024-11-04 10:51:22.522 [wasm      ] [zellij-utils/src/input/layout.rs:109]: Overriding plugin alias 
WARN   |zellij_utils::input::layo| 2024-11-04 10:51:22.522 [wasm      ] [zellij-utils/src/input/layout.rs:109]: Overriding plugin alias 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.528 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction. 
ERROR  |zellij_server::plugins::w| 2024-11-04 10:51:22.531 [async-std/runti] [zellij-server/src/plugins/wasm_bridge.rs:1409]: failed to start plugin 0 for client 1

Caused by:
    0: failed to recover cache dir
    1: WebAssembly translation error
    2: Invalid input WebAssembly code at offset 264822: reference-types not enabled: zero byte expected 
WARN   |zellij_server::route     | 2024-11-04 10:51:22.535 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction. 
ERROR  |zellij_server::plugins::w| 2024-11-04 10:51:22.550 [async-std/runti] [zellij-server/src/plugins/wasm_bridge.rs:1409]: failed to start plugin 1 for client 1

Caused by:
    0: failed to recover cache dir
    1: WebAssembly translation error
    2: Invalid input WebAssembly code at offset 143211: reference-types not enabled: zero byte expected 
ERROR  |zellij_server::plugins   | 2024-11-04 10:52:23.074 [wasm      ] [zellij-server/src/plugins/mod.rs:934]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-11-04 10:52:23.074 [wasm      ] [zellij-server/src/plugins/mod.rs:934]: Plugin with id: 1 not found 

```

**Comment by jscarrott** [comment]
Nuking cache made no difference. I'm just rebuilding from source now.

**Comment by jscarrott** [comment]
Compiling from source works, looks like an issue with the CI?

**Comment by imsnif** [comment]
@bjorn3 - I think we might be hitting https://github.com/bytecodealliance/wasmtime/issues/9130, though I'm unsure why this only happens with the musl builds.

**Comment by imsnif** [comment]
(not even all musl builds btw - for me locally and for the CI e2e tests this works fine)

**Comment by pdanpdan** [comment]
Clearing cache changes nothing.
Log:

```
INFO   |zellij_client            | 2024-11-04 12:04:44.093 [main      ] [zellij-client/src/lib.rs:182]: Starting Zellij client!
INFO   |zellij_server            | 2024-11-04 12:04:44.107 [main      ] [zellij-server/src/lib.rs:522]: Starting Zellij server!
WARN   |zellij_utils::kdl        | 2024-11-04 12:04:44.111 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file
WARN   |zellij_utils::kdl        | 2024-11-04 12:04:44.111 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file
WARN   |zellij_utils::kdl        | 2024-11-04 12:04:44.111 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file
ERROR  |zellij_client::stdin_ansi| 2024-11-04 12:04:44.115 [stdin_handler] [zellij-client/src/stdin_ansi_parser.rs:124]: Failed to open STDIN cache file: Os { code: 2, kind: NotFound, message: "No such file or directory" }
INFO   |zellij_server            | 2024-11-04 12:04:44.117 [main      ] [zellij-server/src/lib.rs:1493]: Compiling plugins using Cranelift
INFO   |zellij_server::plugins   | 2024-11-04 12:04:44.117 [wasm      ] [zellij-server/src/plugins/mod.rs:232]: Wasm main thread starts
WARN   |zellij_server::route     | 2024-11-04 12:04:44.118 [server_router] [zellij-server/src/route.rs:1110]: Server not ready, trying to place instruction in retry queue...
WARN   |zellij_server::route     | 2024-11-04 12:04:44.118 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction.
WARN   |zellij_server::route     | 2024-11-04 12:04:44.123 [server_router] [zellij-server/src/route.rs:1110]: Server not ready, trying to place instruction in retry queue...
WARN   |zellij_server::route     | 2024-11-04 12:04:44.123 [server_router] [zellij-server/src/route.rs:1110]: Server not ready, trying to place instruction in retry queue...
WARN   |zellij_server::route     | 2024-11-04 12:04:44.123 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction.
WARN   |zellij_utils::input::layo| 2024-11-04 12:04:44.126 [wasm      ] [zellij-utils/src/input/layout.rs:109]: Overriding plugin alias
WARN   |zellij_utils::input::layo| 2024-11-04 12:04:44.126 [wasm      ] [zellij-utils/src/input/layout.rs:109]: Overriding plugin alias
WARN   |zellij_server::route     | 2024-11-04 12:04:44.128 [server_router] [zellij-server/src/route.rs:1228]: Server ready, retrying sending instruction.
INFO   |zellij_utils::input::plug| 2024-11-04 12:04:44.136 [async-std/runti] [zellij-utils/src/input/plugins.rs:136]: Plugin 'tab-bar.wasm' exists in the 'PLUGIN DIR' at '/home/pdan/.local/share/zellij/plugins/' but is being ignored
INFO   |zellij_utils::input::plug| 2024-11-04 12:04:44.136 [async-std/runti] [zellij-utils/src/input/plugins.rs:136]: Plugin 'status-bar.wasm' exists in the 'PLUGIN DIR' at '/home/pdan/.local/share/zellij/plugins/' but is being ignored
ERROR  |zellij_server::plugins::w| 2024-11-04 12:04:44.169 [async-std/runti] [zellij-server/src/plugins/wasm_bridge.rs:1409]: failed to start plugin 0 for client 1

Caused by:
    0: failed to recover cache dir
    1: WebAssembly translation error
    2: Invalid input WebAssembly code at offset 449592: reference-types not enabled: zero byte expected
ERROR  |zellij_server::plugins::w| 2024-11-04 12:04:44.169 [async-std/runti] [zellij-server/src/plugins/wasm_bridge.rs:1409]: failed to start plugin 1 for client 1

Caused by:
    0: failed to recover cache dir
    1: WebAssembly translation error
    2: Invalid input WebAssembly code at offset 89302: reference-types not enabled: zero byte expected
INFO   |zellij_client            | 2024-11-04 12:04:45.688 [main      ] [zellij-client/src/lib.rs:584]: Bye from Zellij!
INFO   |zellij_server::plugins   | 2024-11-04 12:04:45.689 [wasm      ] [zellij-server/src/plugins/mod.rs:894]: wasm main thread exits
ERROR  |zellij_server::os_input_o| 2024-11-04 12:04:45.696 [screen    ] [zellij-server/src/os_input_output.rs:917]: Failed to apply cached resizes: failed to send message to pty writer
ERROR  |zellij_utils::errors::not| 2024-11-04 12:04:45.702 [screen    ] [zellij-utils/src/errors.rs:690]: Panic occured:
             thread: screen
             location: At zellij-server/src/lib.rs:1365:18
             message: Program terminates: a fatal error occured

Caused by:
    0: failed to remove client 1
    1: Failed to log and report session state
    2: failed to send message to background jobs
    3: Originating Thread(s)
        1. stdin_handler_thread: AcceptInput
        2. ipc_server: ClientExit
        3. screen_thread: RemoveClient

    4: failed to send message to channel
```

**Comment by bjorn3** [comment]
Adding an entry `'gc'` to the list at https://github.com/zellij-org/zellij/blob/4b806795c3485a247929cbb55db0cdb0b6fd9bed/zellij-server/Cargo.toml#L39-L50 should work.

**Comment by imsnif** [comment]
> Adding an entry `'gc'` to the list at
> should work.

Can confirm this fixes it for me locally - thanks for the super quick assist! Trying to figure out a way to place these binaries in the release.

**Comment by pdanpdan** [comment]
working as expected in v0.41.1, thank you

**Comment by imsnif** [comment]
Thanks for the super quick report @pdanpdan and others. And @bjorn3 for the assist. Looks like the root cause was our CI ignoring the rust-toolchain for some reason. The explicit `gc` wasmtime feature fix from @bjorn3 did the trick - thanks again.

To others: this is fixed in the 0.41.1 patch.
